lib/libzfs_core/libzfs_core.c

```diff
@@ -645,6 +656,90 @@ lzc_send_resume(const char *snapname, const char *from, int fd,
 	    resumeoff, NULL));
 }
 
+unsigned long
+lzc_get_pipe_max()
+{
+	unsigned long max_psize = 65536;
+#ifdef __linux__
+	FILE *procf = fopen("/proc/sys/fs/pipe-max-size", "re");
+
+	if (procf != NULL) {
+		if (fscanf(procf, "%lu", &max_psize) <= 0) {
```

Comment:
You're parsing exactly one field here, so any failure indicates that it remains unchanged. You can drop the if entirely and just do `(void) fscanf();`.

Reply:
Seems reasonable. Sure.

Reply:
Neat, either I did it incorrectly, or no I can't, as I've not seen that much red in the CI in a while.

I knew `--enable-debug` turned on `-Werror`, but I'm surprised I didn't see the warning that's getting promoted to an error without it. Oh well, simple enough...

Reply:
ah, yeah, in that case it'll probably need a dummy variable (and some convincing, potentially)

Reply:
Well, the way I put it back worked on Centos Stream 8 and Debian 10, so hopefully it doesn't get rejected by Ubuntu 18.04 or something.